code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -234,6 +241,87 @@
 		to_chat(user, "<span class='notice'>You insert [W].</span>")
 		show_controls(user)
 
+	else if(istype(W, /obj/item/reagent_containers/syringe/piercing))
+		if(bot_core.allowed(user) && open)
+			if(!piercing)
+				to_chat(user, "<span class='notice'>You replace \the [src] syringe with a diamond-tipped one!</span>")
+				piercing = TRUE
+				qdel(W)
+		if(!open)
+			to_chat(user, "<span class='notice'>\the [src] access pannle is not open!</span>")
+			return
+		else
+			to_chat(user, "<span class='notice'>\the [src] already has a diamond-tipped syringe!</span>")
+
+	else if(istype(W, /obj/item/hypospray/mkii))
+		if(bot_core.allowed(user) && open)
+			if(!hypospray)
+				to_chat(user, "<span class='notice'>You replace \the [src] syringe base with a DeForest Medical MK.II Hypospray!</span>")
+				hypospray = TRUE
+				injection_time = 15 //Half the time half the death!
+				window_name = "Automatic Medical Unit v2.4 ALPHA"
+				qdel(W)
+		if(!open)
+			to_chat(user, "<span class='notice'>\the [src] access pannle is not open!</span>")
+			return
+		else
+			to_chat(user, "<span class='notice'>\the [src] already has a DeForest Medical Hypospray base!</span>")
+
+
+	else if(istype(W, /obj/item/circuitboard/machine/chem_dispenser))
+		if(bot_core.allowed(user) && open)
+			if(!upgraded_dispenser_1)
+				to_chat(user, "<span class='notice'>You add in the board upgrading \the [src] reagent banks!</span>")
+				upgraded_dispenser_1 = TRUE
+				treatment_oxy = /datum/reagent/medicine/salbutamol //Replaces Dex with salbutamol "better" healing of o2
+				qdel(W)
+		if(!open)
+			to_chat(user, "<span class='notice'>\the [src] access pannle is not open!</span>")
+			return
+		else
+			to_chat(user, "<span class='notice'>\the [src] already has a this upgrade!</span>")
+
+	else if(istype(W, /obj/item/circuitboard/machine/cryo_tube))
+		if(bot_core.allowed(user) && open)
+			if(!upgraded_dispenser_2)
+				to_chat(user, "<span class='notice'>You add in the board upgrading \the [src] reagent banks!</span>")
```

Comment:
\the is usually not necessary when doing []'s.
yes, the other code does but that doesn't mean you should too.
leave it alone if you want but it'd be nice if you got rid of it, i'm not too picky about this.